index-advisor/consolidation.mdx

```diff
@@ -0,0 +1,67 @@
+---
+title: 'Index Advisor: Index Consolidation'
+backlink_href: /docs/index-advisor
+backlink_title: 'Index Advisor (in-app in pganalyze)'
+---
+
+The pganalyze Index Advisor includes an **experimental, early access** feature
+to include index consolidation insights (i.e., insights that use `DROP INDEX`
+recommendations in addition to `CREATE INDEX`).
+
+
+## Overview
+
+While indexes can dramatically speed up many queries, they do have some
+performance impact on your database: [write
+overhead](/docs/indexing-engine/index-write-overhead), additional memory needed
+for the index data structure while in use, and disk space. To optimize the
+number of indexes recommended, the Index Advisor considers the full query
+workload on a table rather than recommending indexes specifically for each query
+or each scan. This works well when starting from scratch, but when starting with
+existing indexes, it may lead to over-indexing. Consolidation insights can help.
+
+
+## Limitations
+
+ - Dropping the wrong index can have a catastrophic impact on database
+   performance. We recommend [testing](/docs/index-advisor/test-insights) all
+   Index Advisor insights before implementing them, but this is especially true
+   of consolidation insights.
+ - Index Advisor currently has a limited model of partial indexes (indexes with
+   a `WHERE` clause), and will likely recommend dropping them: these insights
+   should be ignored, or at least reviewed very critically.
```

Comment:
I feel like this is under-selling the impact by being number 2 in a list, and the use of vague terminology ("will likely" instead of "will"). How about:
> Index Advisor doesn't generate index recommendations using `WHERE` clauses because of technical limitations, so when index consolidation is turned on it will recommend dropping all partial indexes. This guidance should be disregarded.

Another option might be to automatically hide removal recommendations of partial indexes in the app, to avoid the potential footgun from team members who haven't read this document or have forgotten the details.

Reply:
Yeah, I think your stronger wording makes sense. I'll update.

We discussed hiding these in-app and I think that makes sense.